public/app/features/dashboard-scene/edit-pane/useDropAndPaste.ts

```diff
@@ -0,0 +1,55 @@
+import { ClipboardEvent, useCallback } from 'react';
+import { useDropzone } from 'react-dropzone';
+
+export function useDropAndPaste() {
+  const onImportFile = useCallback((file?: File) => {
+    if (!file) {
+      return;
+    }
+
+    alert(`Importing file: ${file.name}`);
+  }, []);
+
+  const onPaste = useCallback(
+    (event: ClipboardEvent<HTMLDivElement>) => {
+      const clipboardData = event.clipboardData;
+
+      if (clipboardData.files.length > 0) {
+        // Handle file paste
+        onImportFile(clipboardData.files[0]);
+        return;
+      }
+
+      if (clipboardData.types.includes('text/plain')) {
+        // Handle plaintext paste
+        const text = clipboardData.getData('text/plain');
+        alert(`Pasted text: \n${text}`);
+
+        return;
+      }
+
+      if (clipboardData.types.includes('text/html')) {
+        // Handle HTML paste
+        const html = clipboardData.getData('text/html');
+        alert(`Pasted HTML:\n${html}`);
+      }
+
+      if (clipboardData.types.includes('image/png') || clipboardData.types.includes('image/jpeg')) {
```

Comment:
Full disclosure: this is mostly co-pilot code, so there's probably more supported MIME types